### PR TITLE
[Xamarin.Android.NUnitLite] Import from monodroid/ae417a99

### PR DESCRIPTION
**Pull request by jonpryor**
Xamarin.Android.NUnitLite is a copy of NunitLite, modified for use
with Xamarin.Android and to provide a Xamarin.Android GUI.

### CODE REVIEW

**Comment by jonpryor** [comment]
DO NOT MERGE.

This PR builds from the command line, but Xamarin Studio 6 doesn't like it. Loading the solution results in an error dialog:

> Error while trying to load the project '.../src/Xamarin.Android.NUnitLite/Xamarin.Android.NUnitLite.csproj': Project does not support framework 'MonoAndroid,Version=v6.0'

@radical: Any idea why Xamarin Studio shows the above error message, and how to fix it?

**Comment by jonpryor** [comment]
Xamarin Studio 6.0 will load the project without error if I modify the original [364d8ac3 commit](https://github.com/jonpryor/xamarin-android/commit/364d8ac33ed650b3e2834e97928b6e8dbdcd1942) to include this:

```
diff --git a/src/Xamarin.Android.NUnitLite/Xamarin.Android.NUnitLite.csproj b/src/Xamarin.Android.NUnitLite/Xamarin.Android.NUnitLite.csproj
index 5b5ecdf..b22e7ef 100644
--- a/src/Xamarin.Android.NUnitLite/Xamarin.Android.NUnitLite.csproj
+++ b/src/Xamarin.Android.NUnitLite/Xamarin.Android.NUnitLite.csproj
@@ -7,10 +7,10 @@
     <OutputType>Library</OutputType>
     <RootNamespace>Xamarin.Android.NUnitLite</RootNamespace>
     <AssemblyName>Xamarin.Android.NUnitLite</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
   </PropertyGroup>
   <Import Project="..\..\Configuration.props" />
   <PropertyGroup>
-    <TargetFrameworkVersion>$(AndroidFrameworkVersion)</TargetFrameworkVersion>
     <AndroidSdkDirectory>$(AndroidToolchainDirectory)\sdk</AndroidSdkDirectory>
     <AndroidNdkDirectory>$(AndroidToolchainDirectory)\ndk</AndroidNdkDirectory>
   </PropertyGroup>
@@ -349,6 +349,7 @@
   <Import Project="$(OutputPath)\..\..\..\xbuild\Xamarin\Android\Xamarin.Android.CSharp.targets" />
   <PropertyGroup>
     <ImplicitlyExpandDesignTimeFacades>False</ImplicitlyExpandDesignTimeFacades>
+    <TargetFrameworkIdentifier></TargetFrameworkIdentifier>
   </PropertyGroup>
   <ItemGroup>
     <ProjectReference Include="..\Mono.Android\Mono.Android.csproj">
```

This PR has been updated to include the above patch.

**Comment by jonpryor** [comment]
The latest update [reverts the previous change](https://github.com/xamarin/xamarin-android/pull/13#issuecomment-215837879), and instead adds a `$(ProjectTypeGuids)` MSBuild property so that Xamarin Studio can "properly" load the project.

Pro: Xamarin Studio is able to load the project without generating warnings, and the project can be built from both Xamarin Studio (OS X) and command-line.

Con: I have *no idea at all* what will happen on Linux, e.g. with MonoDevelop. Will `$(ProjectTypeGuids)` be ignored? Will it prevent the project from being loaded? I don't know.

**Comment by jonpryor** [comment]
> ...but I don't know what other ramifications this will imply.

Clearing `$(TargetFrameworkIdentifier)` can break things like the `_GetReferenceAssemblyPaths` target, so that's not a good solution.

**Comment by jonpryor** [comment]
> Con: I have *no idea at all* what will happen on Linux

Fortunately, that's rephrase-able and testable: what happens if we set `$(ProjectTypeGuids)` to a *new* GUID, as produced by `uuidgen`?

Answer: Xamarin Studio *really doesn't like that*, and refuses to load the project:

Xamarin Studio Error

**Comment by jonpryor** [comment]
After some discussion, we're going to drop the "goal" of building *within* MonoDevelop on Linux. The MonoDevelop build system *can* be extended to support loading projects that omit the "flavor" GUID -- closer to what Visual Studio does -- but there is no timeframe for this support, and we kinda *really* need `Xamarin.Android.NUniteLite` so that we can start migrating over unit tests...

Goodbye, Linux IDE build support. I hardly knew ye. (Granted, nothing reliably builds on Linux *anyway*, but now the entire prospect just got a lot harder.)

File: src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITest.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        /// <summary>
+        /// Indicates whether the test can be run using
+        /// the RunState enum.
+        /// </summary>
+		RunState RunState { get; set; }
+
+        /// <summary>
+        /// Count of the test cases ( 1 if this is a test case )
+        /// </summary>
+		int TestCaseCount { get; }
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITest.cs]: @jonpryor Indentation is off :(

File: src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITestCaseData.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        /// <summary>
+        /// Gets the name to be used for the test
+        /// </summary>
+        string TestName { get; }
+		
+		/// <summary>
+		/// Gets the RunState for this test case.
+		/// </summary>
+		RunState RunState { get; }
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITestCaseData.cs]: @jonpryor ditto.

File: src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITestFilter.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        bool IsEmpty { get; }
+
+		/// <summary>
+		/// Determine if a particular test passes the filter criteria. Pass
+		/// may examine the parents and/or descendants of a test, depending
+		/// on the semantics of the particular filter
+		/// </summary>
+		/// <param name="test">The test to which the filter is applied</param>
+		/// <returns>True if the test passes the fFilter, otherwise false</returns>
+		bool Pass( ITest test );
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/Api/ITestFilter.cs]: @jonpryor +2

File: src/Xamarin.Android.NUnitLite/NUnitLite/Api/TestOutput.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	{
+		string text;
+		TestOutputType type;
+
+        /// <summary>
+        /// Construct with text and an ouput destination type
+        /// </summary>
+        /// <param name="text">Text to be output</param>
+        /// <param name="type">Destination of output</param>
+		public TestOutput(string text, TestOutputType type)
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/Api/TestOutput.cs]: @jonpryor same. is it a tab's vs space's issue?

File: src/Xamarin.Android.NUnitLite/NUnitLite/AssertionHelper.cs [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+        /// <summary>
+		/// Returns a ListMapper based on a collection.
+		/// </summary>
+		/// <param name="original">The original collection</param>
+		/// <returns></returns>
+		public ListMapper Map( ICollection original )
+		{
+			return new ListMapper( original );
+		}
+		#endregion
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/AssertionHelper.cs]: @jonpryor indentation :(

File: src/Xamarin.Android.NUnitLite/NUnitLite/Attributes/ExplicitAttribute.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+namespace NUnit.Framework
+{
+	/// <summary>
+	/// ExplicitAttribute marks a test or test fixture so that it will
+	/// only be run if explicitly executed from the gui or command line
+	/// or if it is included by use of a filter. The test will not be
+	/// run simply because an enclosing suite is run.
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Class|AttributeTargets.Method|AttributeTargets.Assembly, AllowMultiple=false, Inherited=false)]
+    public class ExplicitAttribute : NUnitAttribute, IApplyToTest
```
Comment (dellis1972) [inline src/Xamarin.Android.NUnitLite/NUnitLite/Attributes/ExplicitAttribute.cs]: @jonpryor this looks like a tabs vs spaces issue. Did we decide on a code format for the OSS stuff? I assume its VS standard?

**Comment by dellis1972** [comment]
@jonpryor This looks ok for the most part. There are a bunch of indentation issues probably as a result of the original code being a mix of spaces/tabs.
I think we should fix them up. We should probably add a XS code formatting rule to the projects as well so that the IDE's will use the right formatting. We have one of those in MonoGame so I'll see if I can bring that over. I assume we are using VS code formatting.

**Comment by dellis1972** [comment]
PR #22 adds the code formatting xml file

**Pull request merged**